cpp/src/parquet/arrow/writer.cc

```diff
@@ -518,6 +518,36 @@ class FileWriterImpl : public FileWriter {
     return Status::OK();
   }
 
+  Status WriteRecordBatch(const ::arrow::RecordBatch& batch) override {
+    auto schema = batch.schema();
+    auto size = batch.num_rows();
+
+    if (!schema->Equals(*schema_, false)) {
+      return Status::Invalid("RecordBatch schema does not match this writer's. batch:'",
+                             schema->ToString(), "' this:'", schema_->ToString(), "'");
+    }
+
+    RETURN_NOT_OK(NewRowGroup(size));
```

Comment:
I don't think we should be encouraging this. Parquet row groups are supposed to be large, while RecordBatches in general are going to be smaller. Guidance in Hadoop was to have row groups be approximately ~1GB (or around the size of an HDFS block) where systems like Dremio have relatively small record batches of 64K rows max. So a row group is in general going to be composed from many row groups

Reply:
We can remove this function and make it internal for testing only. I wasn't aware that RowGroups were limited, my mental model is that they were closer to what Dremio is doing ~64k.